src/main/java/org/prebid/server/gdpr/vendorlist/VendorListService.java

```diff
@@ -246,7 +246,7 @@ private static boolean isValid(VendorList vendorList) {
                 && vendorList.getLastUpdated() != null
                 && CollectionUtils.isNotEmpty(vendorList.getVendors())
                 && vendorList.getVendors().stream()
-                .allMatch(vendor -> vendor != null && vendor.getId() != null && vendor.getPurposeIds() != null);
+                .allMatch(vendor -> vendor != null && vendor.getId() != null && vendor.combinedPurposes() != null);
```

Comment:
the `combinedPurposes(...)` method never returns null result, so we should replace condition
```
vendor.combinedPurposes() != null
```
with:
```
(vendor.getPurposeIds() != null || vendor.getLegIntPurposeIds() != null)
```